cloudberry/zion/src/test/scala/edu/uci/ics/cloudberry/zion/model/impl/SQLGeneratortest.scala

```diff
@@ -0,0 +1,786 @@
+      val result = parser.generate(query, Map(TwitterDataSetForSQL -> twitterSchemaForSQL))
+      removeEmptyLine(result) must_== unifyNewLine(
+        """
+          |select hour(t.`create_at`) as `hour`,count(*) as `count`
+          |from `twitter_ds_tweet` t
+          |where t.`geo_tag.stateID` in ( 37,51,24,11,10,34,42,9,44 )
+          |group by `hour`
+          | """.stripMargin.trim)
+    }
+
+    //22
+    "translate a text contain + time + geo id set filter and group by time + spatial cube" in {
+      val filter = Seq(textFilter, timeFilter, stateFilter)
+      val group = GroupStatement(Seq(byHour, byGeoState), Seq(aggrCount))
+      val query = new Query(TwitterDataSetForSQL, Seq.empty, Seq.empty, filter, Seq.empty, Some(group), None)
+      val result = parser.generate(query, Map(TwitterDataSetForSQL -> twitterSchemaForSQL))
+      removeEmptyLine(result) must_== unifyNewLine(
+        """
+          |select hour(t.`create_at`) as `hour`,t.`geo_tag.stateID` as `state`,count(*) as `count`
+          |from `twitter_ds_tweet` t
+          |where match(t.`text`) against ('+zika +virus' in boolean mode) and t.`create_at` >= '2016-01-01 00:00:00.000' and t.`create_at` < '2016-12-01 00:00:00.000' and t.`geo_tag.stateID` in ( 37,51,24,11,10,34,42,9,44 )
+          |group by `hour`,`state`
+          | """.stripMargin.trim)
+    }
+
+    //23
+    "translate a text contain + time + geo id set filter and sample tweets" in {
+      val filter = Seq(textFilter, timeFilter, stateFilter)
+      val query = new Query(TwitterDataSetForSQL, Seq.empty, Seq.empty, filter, Seq.empty, None, Some(selectRecent))
+      val result = parser.generate(query, Map(TwitterDataSetForSQL -> twitterSchemaForSQL))
+      removeEmptyLine(result) must_== unifyNewLine(
+        """
+          |select t.`place.full_name` as `place.full_name`,t.`place.bounding_box` as `place.bounding_box`,t.`place.country_code` as `place.country_code`,t.`user.friends_count` as `user.friends_count`,t.`user.description` as `user.description`,t.`favorite_count` as `favorite_count`,t.`geo_tag.countyID` as `geo_tag.countyID`,t.`user.location` as `user.location`,t.`place.type` as `place.type`,t.`geo_tag.cityName` as `geo_tag.cityName`,t.`user.id` as `user.id`,t.`geo_tag.stateName` as `geo_tag.stateName`,t.`geo_tag.cityID` as `geo_tag.cityID`,t.`is_retweet` as `is_retweet`,t.`text` as `text`,t.`user.screen_name` as `user.screen_name`,t.`retweet_count` as `retweet_count`,t.`place.country` as `place.country`,t.`in_reply_to_user` as `in_reply_to_user`,t.`user.statues_count` as `user.statues_count`,t.`id` as `id`,t.`coordinate` as `coordinate`,t.`place.id` as `place.id`,t.`in_reply_to_status` as `in_reply_to_status`,t.`geo_tag.stateID` as `geo_tag.stateID`,t.`create_at` as `create_at`,t.`user.create_at` as `user.create_at`,t.`place.name` as `place.name`,t.`lang` as `lang`,t.`user.lang` as `user.lang`,t.`user.name` as `user.name`,t.`geo_tag.countyName` as `geo_tag.countyName`
```

Comment:
Thanks for remind! I've fixed this bug. 
It seems that sparkSQL also has this bug unfixed, do you want me to fix it?